### DISCUSSION TITLE
h3 error "Cannot set headers after they are sent to the client"

### DISCUSSION THREAD

**Discussion by nikkioon** [nuxt/nuxt#17651] [Questions] [answered]
Hi,

Im Using
"nuxt": "^3.0.0-rc.1",
"@pinia/nuxt": "^0.1.8",
"pinia": "^2.0.13"

Sometimes I get the error in my IDE:
<img width="680" alt="2022-04-27 um 08 06 14" src="https://user-images.githubusercontent.com/104486336/165455554-7396aaa8-1ce4-4fe7-a5cb-3c3164d8ae41.png">

So far I have not been able to find out where this error comes from or reproduce it.

Does anyone have any idea what triggers it or which part of Nuxt/Pinia is responsible?

**Comment by viii47** [comment]
You're trying to set values into the HTTP Header when the operation is already through (too late) or a different function/module/server API already handled it. 

> Does anyone have any idea what triggers it or which part of Nuxt/Pinia is responsible?

You need to check any part of your custom code (I'd start in plugins, server/api, components) that you've implemented that sets any http header values (for example cookies) on client. Beware that H3 cannot handle multi-calls at once -> you need an interceptor or module for this (like axios).

**Comment by nikkioon** [comment, marked as answer]
have found out what was responsible for the error.
In Layouts/default.vue I have set the favicon with a link tag
`<Link rel="icon" type="image/x-icon" href="/favicon.ico"/>`
since i moved the favicon link to the nuxt.config the error does not occur anymore.
